core/src/main/java/nl/nn/adapterframework/filesystem/FileSystemActor.java

```diff
@@ -508,7 +508,12 @@ private void writeContentsToFile(OutputStream out, Message input, ParameterValue
 		} else if (contents instanceof byte[]) {
 			out.write((byte[])contents);
 		} else if (contents instanceof String) {
-			out.write(((String) contents).getBytes(StreamUtil.DEFAULT_INPUT_STREAM_ENCODING));
+			if(StringUtils.isNotEmpty(getCharset())) {
+				out.write(((String) contents).getBytes(getCharset()));
+			} else {
+				out.write(((String) contents).getBytes());
```

Comment:
This should remain `StreamUtil.DEFAULT_INPUT_STREAM_ENCODING` else the platforms default charset will be used, on windows this is `windows-1252`.